lib/GameConstants.h

```diff
@@ -745,6 +745,27 @@ DLL_LINKAGE std::ostream & operator<<(std::ostream & os, const ETerrainType acti
 
 ID_LIKE_OPERATORS_DECLS(ETerrainType, ETerrainType::EETerrainType)
 
+class DLL_LINKAGE EPathfindingLayer
+{
+public:
+	enum EEPathfindingLayer
+	{
+		WRONG = -2, AUTO = -1, LAND = 0, SAIL = 1, WATER, AIR, NUM_LAYERS

```

Comment:
I'm really like idea to rename them for sure and your naming isn't bad, though I'm don't like "WALK". Not going to change names for now as I'm have few ideas of how I can possibly improve pathfinder design by splitting ruleset of each layer into separate class, etc.

Though if I'm decide to keep it as is in the end then I'll clearly rename layers.